drivers/adc/ad7768-1/ad77681.c

```diff
@@ -1169,6 +1170,139 @@ int32_t ad77681_apply_gain(struct ad77681_dev *dev,
 	return ret;
 }
 
+/**
+ * Upload sequence for Programmamble FIR filter
+ * @param dev - The device structure.
+ * @param coeffs - Pointer to the desired filter coefficients array to be written
+ * @param num_coeffs - Count of active filter coeffs
+ * @return 0 in case of success, negative error code otherwise.
+ */
+int32_t ad77681_programmable_filter(struct ad77681_dev *dev,
+				    const float *coeffs,
+				    uint8_t num_coeffs)
+{
+	uint8_t coeffs_buf[4], coeffs_index, check_back = 0, i, address;
+	uint32_t twait;
+	int32_t twos_complement, ret;
+	const uint8_t coeff_reg_length = 56;
+
+	/* Specific keys in the upload sequence */
+	const uint8_t key1 = 0xAC, key2 = 0x45, key3 = 0x55;
+	/* Scaling factor for all coefficients 2^22 */
+	const float coeff_scale_factor = (1 << 22);
+	/* Wait time in uS necessary to access the COEFF_CONTROL and */
+	/* COEFF_DATA registers. Twait = 512/MCLK */
+	twait = (uint32_t)(((512.0) / ((float)(dev->mclk))) * 1000.0) + 1;
+
+
+	/* Set Filter to FIR */
+	ret = ad77681_spi_write_mask(dev,
+				     AD77681_REG_DIGITAL_FILTER,
+				     AD77681_DIGI_FILTER_FILTER_MSK,
+				     AD77681_DIGI_FILTER_FILTER(AD77681_FIR));
+
+	/* Write the first access key to the ACCESS_KEY register */
+	ret |= ad77681_spi_write_mask(dev,
```

Comment:
We would like to avoid this way of reacting to error code. I know there are similar examples in no-OS code to what you did but we're working on improving things.

What you should do after each call is something like:
```
if (ret < 0)
    return ret;
```

Reply:
Can I do this? 

	if ((check_back != 1) | (ret < 0))
		return FAILURE;

I also need to check the key bit if it is equal to 1 before I could start the coefficient writes.
![image](https://user-images.githubusercontent.com/65901998/88175429-6e6c4880-cc58-11ea-8b8a-bc402f703d56.png)

Reply:
Of course you can combine, but in this case I would do it like this:
```
if ((ret < 0) || (check_back != 1))
	return FAILURE;
```
Placing the ret test first, because it doesn't make sense to test the parameters that the function outputs through pointers if the return of the function is an error.

Reply:
Thanks. Done implementing changes.